pkgs/development/lisp-modules/default.nix

```diff
@@ -0,0 +1,299 @@
+# TODO:
+# - faster build by using lisp with preloaded asdf?
+# - dont include java libs unless abcl?
+# - dont use build-asdf-system to build lispWithPackages?
+# - make the lisp packages overridable? (e.g. buildInputs glibc->musl)
+# - build asdf with nix and use that instead of one shipped with impls
+#   (e.g. to fix build with clisp - does anyone use clisp?)
+# - claspPackages ? (gotta package clasp with nix first)
+# - hard one: remove unrelated sources ( of systems not being built)
+# - figure out a less awkward way to patch sources
+#   (have to build from src directly for SLIME to work, so can't just patch sources in place)
+
+{ pkgs
+, lib
+, stdenv
+, ... }:
+
+let
+
+  inherit (lib)
+    length
+    filter
+    foldl
+    unique
+    id
+    concat
+    concatMap
+    mutuallyExclusive
+    findFirst
+    remove
+    setAttr
+    getAttr
+    hasAttr
+    attrNames
+    attrValues
+    filterAttrs
+    mapAttrs
+    splitString
+    concatStringsSep
+    concatMapStringsSep
+    replaceStrings
+    removeSuffix
+    hasInfix
+    optionalString
+    makeBinPath
+    makeLibraryPath
+    makeSearchPath
+    recurseIntoAttrs
+  ;
+
+  inherit (builtins)
+    head
+    tail
+    elem
+    split
+    storeDir;
+
+  inherit (pkgs)
+    substituteAll;
+
+  # Stolen from python-packages.nix
+  # Actually no idea how this works
+  makeOverridableLispPackage = f: origArgs:
+    let
+      ff = f origArgs;
+      overrideWith = newArgs: origArgs // (if pkgs.lib.isFunction newArgs then newArgs origArgs else newArgs);
+    in
+      if builtins.isAttrs ff then (ff // {
+        overrideLispAttrs = newArgs: makeOverridableLispPackage f (overrideWith newArgs);
+      })
+      else if builtins.isFunction ff then {
+        overrideLispAttrs = newArgs: makeOverridableLispPackage f (overrideWith newArgs);
+        __functor = self: ff;
+      }
+      else ff;
+
+  buildAsdf = { asdf, pkg, program, flags, faslExt }:
+    stdenv.mkDerivation {
+      inherit (asdf) pname version;
+      dontUnpack = true;
+      buildPhase = ''
+        cp -v ${asdf}/lib/common-lisp/asdf/build/asdf.lisp asdf.lisp
+        ${pkg}/bin/${program} ${flags} < <(echo '(compile-file "asdf.lisp")')
```

Comment:
This does not look cross friendly

Reply:
If we mostly use SBCL (not really cross friendly), and ECL (… quite limited cross-friendliness, I would say), maybe it does not matter.

Reply:
I am not experienced with cross compiling amd would need some help

Reply:
I would say it's fine to just document lack of cross-compilation support, as upstreams do not support it uniformly.